python/pdo/submitter/ccf/ccf_submitter.py

```diff
@@ -0,0 +1,589 @@
+        contract_enclave_id,
+        enclave_signature,
+        contract_id,
+        message_hash,
+        current_state_hash,
+        contract_code_hash,
+        **extra_params):
+
+        tx_method = "ccl_update"
+        verb = "init"
+        previous_state_hash = [];
+        dependency_list = [];
+
+        tx_params = PayloadBuilder.build_initialize_contract_state_transaction_from_data(
+            verb,
+            contract_enclave_id,
+            crypto.base64_to_byte_array(enclave_signature),
+            contract_id,
+            current_state_hash,
+            previous_state_hash,
+            dependency_list,
+            message_hash,
+            contract_code_hash,
+            self.pdo_signer.signing_key,
+            self.pdo_signer.verifying_key,
+            nonce = channel_keys
+            )
+        try:
+            response = self.submit_rpc_to_ccf(tx_method, tx_params)
+            if response.result: # result will be True for successful init transaction
+                return tx_params['nonce'] # this will represent the transaction id
+            else:
+                raise Exception(response.error['message'])
+        except Exception as e:
+            raise
+
+# -----------------------------------------------------------------
+    def ccl_update(self,
+        channel_keys,
+        contract_enclave_id,
+        enclave_signature,
+        contract_id,
+        message_hash,
+        current_state_hash,
+        previous_state_hash,
+        dependency_list,
+        **extra_params):
+
+        tx_method = "ccl_update"
+        verb = "update"
+        contract_code_hash =[]
+
+        dependencies = []
+        for dependency in dependency_list :
+            temp = dict()
+            temp['contract_id'] = dependency['contract_id']
+            temp['state_hash'] = crypto.base64_to_byte_array(dependency['state_hash'])
+            temp['state_hash_for_sign'] = dependency['state_hash']
+            dependencies.append(temp)
+
+        tx_params = PayloadBuilder.build_update_contract_state_transaction_from_data(
+            verb,
+            contract_enclave_id,
+            crypto.base64_to_byte_array(enclave_signature),
+            contract_id,
+            current_state_hash,
+            previous_state_hash,
+            dependencies,
+            message_hash,
+            contract_code_hash,
+            nonce = channel_keys
+            )
+
+        try:
+            response = self.submit_rpc_to_ccf(tx_method, tx_params)
+            if response.result: # result will be True for successful init transaction
+                return tx_params['nonce'] #this will represent the transaction id
+            else:
+                raise Exception(response.error['message'])
+        except Exception as e:
+            raise
+
+# -----------------------------------------------------------------
+    def get_enclave_info(self,
+        enclave_id):
+
+        tx_method = "verify_enclave_registration"
```

Comment:
any reason to use `verify` in the transaction name here?
`get_*` is more appropriate since you actually grab a lot of data.

Reply:
thank you, please see answer above.